core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -175,6 +180,21 @@ public ConnectorToSqlExpressionTranslator(Session session, PlannerContext planne
             if (OR_FUNCTION_NAME.equals(call.getFunctionName())) {
                 return translateLogicalExpression(LogicalExpression.Operator.OR, call.getArguments());
             }
+            if (NOT_FUNCTION_NAME.equals(call.getFunctionName())) {
+                if (call.getArguments().size() == 1) {
+                    if (call.getArguments().get(0) instanceof Call) {
+                        Call innerCall = (Call) call.getArguments().get(0);
+                        if (innerCall.getFunctionName().equals(IS_NULL_FUNCTION_NAME) && innerCall.getArguments().size() == 1) {
+                            return translateIsNotNull(innerCall.getArguments().get(0));
+                        }
+                    }
+
+                    return translateNot(call.getArguments().get(0));
```

Comment:
`call.getArguments().get(0)` is used thrice in this block, capture this as a variable
```
argument = getOnlyElement(call.getArguments());
```